src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateClone = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        Object.assign(stateClone, action.extraData);
+
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of action.keysToRemove) {
+          delete stateClone[key];
+        }
+
+        break;
+      }
+
+      case 'clear': {
+        for (const key in stateClone) {
+          delete stateClone[key];
+        }
+
+        break;
+      }
+    }
```

Comment:
A 'default' case should be added to the switch statement for error handling, to manage or report an action type that is not recognized.